### PR TITLE
chore(pixtral): emit block attention mask when using flash attention

### PR DESCRIPTION
**Pull request by starcatmeow**
# What does this PR do?

Since flash_attention_2 relies solely on position_ids, emitting the block attention mask avoids unnecessary memory usage and prevents OOM on large inputs.

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?

## Who can review?

- @uminaty 
- @amyeroberts
- @qubvel

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @zucchini-nlp for VLMs I think

**Review by zucchini-nlp — APPROVED** [review]
Thanks, makes sense. We can delete the line which sets `attention_mask = None` in the attention module now, since it's guaranteed to be `None`

**Review by uminaty — COMMENTED** [review]
LGTM, Maybe it makes [this line](https://github.com/starcatmeow/transformers/blob/fcd4a54cb5d77303ddc2e376351161c5410d1b39/src/transformers/models/pixtral/modeling_pixtral.py#L217) useless. (Edit : didn't see the previous comment!)

**Comment by starcatmeow** [comment]
@zucchini-nlp True, I removed that
